src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
I added a test for @JonHanna's expression. Let's see what the CI build has to say (locally it works as expected).

It's still so weird, looking at it side by side...

    Console.WriteLine(float.IsInfinity(float.MaxValue * 2.0f)); // true
    Console.WriteLine(double.IsInfinity(float.MaxValue * 2.0f)); // false